apps/dapp/src/components/Pages/Core/DappPages/Dashboard/Table/TxnHistoryTable.tsx

```diff
@@ -1,153 +1,199 @@
+      res.strategies.map(s=> txCountTotal += s.transactionCount);
+      setTotalPages(Math.ceil(txCountTotal / rowsPerPage));
+      setBlockNumber(res._meta.block.number);
+    })();
+  }, [dashboardType, rowsPerPage, filter, restartTable]);
+
+  const fetchTransactions = useCallback(
+    async (strategyType: StrategyType): Promise<Transactions> => {
+      
+      const strategyQuery = strategyType === 'All' ? `` : `where: { name: "${strategyType}" }`;
+      const blockNumberQueryParam = blockNumber > 0 ? `block: { number: ${blockNumber} }` : ``;
+      let strategyAndBlockQuery = '';
+      if (blockNumberQueryParam.length > 0 || strategyQuery.length> 0){
+        strategyAndBlockQuery = `(${blockNumberQueryParam} ${strategyQuery})`
+      }
+      
+      const paginationQuery = `skip: ${(currentPage - 1) * rowsPerPage} first: ${rowsPerPage}`;
+      
+      const dateNowSecs = Math.round( Date.now() / 1000);
+      const filterQuery = `where: { timestamp_gt: ${ dateNowSecs - txHistoryFilterTypeToSeconds(filter)} }`;
+
+      const { data: res } = await fetchGenericSubgraph<FetchTxnsResponse>(
+        env.subgraph.templeV2,
+        `{
+          strategies${strategyAndBlockQuery} {
+            name
+            id
+            transactions(orderBy: timestamp, orderDirection: desc ${paginationQuery} ${filterQuery}) {
+              hash
+              amount
+              amountUSD
+              id
+              from
+              kind
+              timestamp
+            }
+          }
+        }`
+      );
+      if (!res) return [];
+      const txns: Transactions = [];
+      for (const strategy of res.strategies) {
+        strategy.transactions.map((t) => txns.push(t));
+      }
+      return txns;
+    },
+    [blockNumber, currentPage, rowsPerPage, filter]
+  );
 
-  const lastRow = currentPage * rowsPerPage;
-  const firstRow = lastRow - rowsPerPage;
-  const totalPages = Math.ceil(data.length / rowsPerPage);
-  const currentData = data.slice(firstRow, lastRow);
-
-  // Fetch data
+  // Fetch strategies tx data
   useEffect(() => {
-    const fetchData = async () => {
+    (async () => {
+      setIsLoading(true);
       try {
-        // Replace with your API endpoint
-        // const response = await fetch('https://api.yourdomain.com/data-endpoint');
-        // const result = await response.json();
-
-        // setData(result.data);
-        // TODO: USE filter
-        console.log('filter', filter);
+        const txns = await fetchTransactions(dashboardTypeToStrategyType(dashboardType));
         setData(
-          new Array(100).fill(100).map((e, i) => {
+          txns.map((tx) => {
+            const amount = Number(Number(tx.amount).toFixed(2));
             return {
-              date: '2021-09-01',
-              debit: i,
-              credit: 100,
+              date: format(new Date(Number(tx.timestamp) * 1000), 'yyyy-MM-dd'),
+              dToken: 'dUsd', // TODO: update dynamically from strategy
```

Comment:
@medariox do we have a way to get the `debtToken` symbol from each strategy?

Reply:
how do I related each transaction to a debt token, e.g. TLC strategy has both `dUSD` & `dTEMPLE`, and there are 15 transactions, I imagine the amount we show in the transactions is in usd amounts, therefore it'll be always `dUSD`?

Reply:
All transactions have `amounts` and `amountsUSD` (amountsUSD is 0 for Sepolia because of the missing price providers on the chain). See [this example](https://api.studio.thegraph.com/query/520/v2-sepolia/version/latest/graphql?query=%7B%0A++strategies%28where%3A+%7Bname%3A+%22TlcStrategy%22%7D%29+%7B%0A++++name%0A++++id%0A++++debtTokens+%7B%0A++++++name%0A++++++id%0A++++++symbol%0A++++%7D%0A++++transactions%28orderBy%3A+timestamp%2C+orderDirection%3A+desc%29+%7B%0A++++++amount%0A++++++amountUSD%0A++++++from%0A++++++to%0A++++++hash%0A++++++kind%0A++++%7D%0A++%7D%0A++_meta+%7B%0A++++block+%7B%0A++++++number%0A++++%7D%0A++%7D%0A%7D).

Reply:
so... each transaction can be both dUSD and dTEMPLE? if so, which one should we show, sound like maybe there could be an option where user can select one or another? or should we have both columns, one for each debt token amount?

Reply:
@medariox In that example query, I think you also need to add the `token` that the transaction was for. 

Example tx's:

- ramos strategy borrows DAI, get issued dUSD
- ramos strategy repays DAI, dUSD is burned
- ramos strategy borrows TEMPLE get issued dTEMPLE
- ramos strategy repays TEMPLE, dTEMPLE is burned

So in this case, we need to know what token was borrowed or repaid

Reply:
Alright, I see what you mean now. I didn't think we wanted to show that since we have the USD amounts, but in hindsight, it's good info that can be added easily. I will add a `token` field to the Transaction entity later today.

Reply:
I've added the `token` field to the Strategy / TLC transactions entity as requested, see [this example query](https://api.studio.thegraph.com/query/520/v2-sepolia/version/latest/graphql?query=%7B%0A++strategies%28where%3A+%7Bname%3A+%22TlcStrategy%22%7D%29+%7B%0A++++name%0A++++id%0A++++debtTokens+%7B%0A++++++name%0A++++++id%0A++++++symbol%0A++++%7D%0A++++transactions%28orderBy%3A+timestamp%2C+orderDirection%3A+desc%29+%7B%0A++++++token+%7B%0A++++++++id%0A++++++++name%0A++++++++symbol%0A++++++%7D%0A++++++amount%0A++++++amountUSD%0A++++++from%0A++++++to%0A++++++hash%0A++++++kind%0A++++%7D%0A++%7D%0A++_meta+%7B%0A++++block+%7B%0A++++++number%0A++++%7D%0A++%7D%0A%7D).

Reply:
thx sorted out